src/plugins/dataflow-tool/components/dataflow-program.tsx

```diff
@@ -1094,10 +1109,44 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     } else {
       this.stores.serialDevice.setSerialNodesCount(0);
     }
+
+    if (serialNodesCt > 0 && !this.stores.serialDevice.hasPort()){
+      this.postSerialModal();
+    }
   }
 
-  private updateNodeChannelInfo = (n: Node) => {
+  private sendDataToSerialDevice(n: Node){
+    if (!isNaN(n.data.nodeValue as number)){
```

Comment:
```suggestion
    if (isFinite(n.data.nodeValue as number)){
```
Avoids the implied double negative ("not not-a-number") and catches infinities as well.